### ISSUE TITLE
Add a tag system

### ISSUE THREAD

**Issue by SleepySwords** [SleepySwords/do_todo#14]
Add a tag or label system that is easily able to filter and customise. TUI's support Unicode emojis. Take advantage of that to design pleasant-looking tags.

**Comment by Froloket64** [comment]
What do you think about, instead of forcing the user to create a new tag, letting the user either choose an already created tag (all of which would be stored somewhere), or create a new one (storing it)?

And as for removing tags, I think we need to rewrite the dialog to only show tags _actually_ attached to the selected task.

**Comment by SleepySwords** [comment]
> What do you think about, instead of forcing the user to create a new tag, letting the user either choose an already created tag (all of which would be stored somewhere), or create a new one (storing it)?

Isn't it already like this? I did change it so one button would flip a tag and another would create/remove tags which could cause confusion, probably should revert it to a single button for both flipping and creating/removing.

**Comment by Froloket64** [comment]
Yeah, but it still shows **all** the tags there are, not only the ones attached to the current task. Besides that, I think we should merge them into one key again (and it's not even in the help menu, which is another reason to centralize key actions creation - #44 :) )

**Comment by SleepySwords** [comment]
> (and it's not even in the help menu, which is another reason to centralize key actions creation - https://github.com/SleepySwords/do_todo/issues/44 :) )

Whoops, that's my bad, I did have a plan that `available_actions` would replace the actions in `key_pressed` to avoid this but didn't really get around to it.

**Comment by Froloket64** [comment]
I think it should be like this:
A user can create a number of tags. Then, they can take a tag out of that collection and attach it to a given task. If they want a new tag, they can create it for the future use for their tasks. If they want to remove a tag from a task, they can select a given task, press a key and select one of the tags attached to this specific task. If they want to edit a tag (say, change its color), they can just press another key (or select a different entry in the same menu, not sure yet) and modify the tag, which will change its state on all tasks, which it's attached to.

> Is there going to be a separate key for removal/addition? Or is there going to be a nested prompt?

I say use one key for everything (creation, attachment and disattachment), except for task editing (which can use <kbd>T</kbd>)

**Comment by SleepySwords** [comment]
The thing I'm still a bit confused about is if the tags are added and removed in the same dialogue box or different boxes.

**Comment by Froloket64** [comment]
Added, removed and created in the same one (nested), and edited in a different one.

**Comment by SleepySwords** [comment]
Alright cool
